consensus/blake3pow/poem.go

```diff
@@ -268,12 +268,27 @@ func (blake3pow *Blake3pow) CalcRank(chain consensus.ChainHeaderReader, header *
 	return 0, nil
 }
 
-func (blake3pow *Blake3pow) CheckIfValidWorkShare(workShare *types.WorkObjectHeader) bool {
-	workShareMinTarget, err := consensus.CalcWorkShareThreshold(workShare)
-	if err != nil {
-		blake3pow.logger.Error(err)
+func (blake3pow *Blake3pow) CheckIfValidWorkShare(workShare *types.WorkObjectHeader) types.WorkShareValidity {
+	if blake3pow.CheckWorkThreshold(workShare, params.WorkSharesThresholdDiff) {
+		return types.Valid
+	} else if blake3pow.CheckWorkThreshold(workShare, blake3pow.config.WorkShareThreshold) {
```

Comment:
this value is dynamic based on the workshares threshold diff

Reply:
Its calculated now when reading from the flag